rust/private/rustc.bzl

```diff
@@ -210,33 +210,14 @@ def _get_linker_and_args(ctx, rpaths):
 
     return ld, link_args, link_env
 
-def rustc_compile_action(
+def _rustc_compile_inputs(
         ctx,
         toolchain,
         crate_info,
-        output_hash = None,
-        rust_flags = []):
-    """
-    Constructs the rustc command used to build the current target.
-
-    Returns:
-      List[Provider]: A list of the following providers:
-                     - CrateInfo: info for the crate we just built; same as `crate_info` parameter.
-                     - DepInfo: The transitive dependencies of this crate.
-                     - DefaultInfo: The output file for this crate, and its runfiles.
-    """
-    output_dir = crate_info.output.dirname
-
-    dep_info, build_info = collect_deps(
-        crate_info.deps,
-        crate_info.aliases,
-        toolchain,
-    )
-
+        dep_info,
+        build_info):
```

Comment:
I considered this, but decided not to, for a few reasons:

- collect_deps is doing significantly more work than "getattr" -- rather than being a simple lookup, it's O(N) with the number of dependencies, which I'd like to avoid repeating multiple times.
- In comparison, the linker script is basically "get a variable", rather than "do any work".

WDYT?

Reply:
sounds fine.